pylxd/models/storage_pool.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2016 Canonical Ltd
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+from pylxd.models import _model as model
+
+
+class StoragePool(model.Model):
+    """A LXD storage_pool."""
+    name = model.Attribute()
+    driver = model.Attribute()
+    description = model.Attribute()
+    used_by = model.Attribute()
+    config = model.Attribute()
+    managed = model.Attribute()
+
+    @classmethod
+    def get(cls, client, name):
+        """Get a storage_pool by name."""
+        response = client.api.storage_pools[name].get()
+
+        storage_pool = cls(client, **response.json()['metadata'])
+        return storage_pool
+
+    @classmethod
+    def all(cls, client):
+        """Get all storage_pools."""
+        response = client.api.storage_pools.get()
+
+        storage_pools = []
+        for url in response.json()['metadata']:
+            name = url.split('/')[-1]
+            storage_pools.append(cls(client, name=name))
+        return storage_pools
+
+    @property
+    def api(self):
```

Comment:
Refactoring note for the future.  This method is (nearly) identical in every Model derived class. It's possible other methods here are too.  Not to be changed for the commit though.